src/lib.rs

```diff
@@ -1,1936 +1,120 @@
-                    self.inner.test_complex_where_clause(first, second, third)
-                }
-
-                #[inline]
-                fn test_complex_where_clause_with_return<T, U, V>(self) -> V
-                    where T: SomeTrait + AnotherTrait,
-                          U: Yet<T> + AnotherTrait,
-                          V: StillAnotherTrait<T, U> + ?Sized {
-                    self.inner.test_complex_where_clause_with_return()
-                }
-            }
-        }
+            Ok(DelegatedSegment { delegator, methods })
+        })
     }
+}
 
-    #[test]
-    fn test_comments() {
-        assert_delegation! {
-            {
-                target self.inner {
-                    // simple comment
-                    fn test_simple_comment(self);
-
-                    // multiple
-                    // simple
-                    // comments
-                    fn test_multiple_simple_comments(self);
-
-                    /// doc comment
-                    fn test_doc_comment(self);
-
-                    /// multiple
-                    /// doc
-                    /// comments
-                    fn test_multiple_doc_comments(self);
-
-                    //! bang comment
-                    fn test_bang_comment(self);
-
-                    //! multiple
-                    //! bang
-                    //! comments
-                    fn test_multiple_bang_comments(self);
-                }
-            },
-
-            {
-                #[inline]
-                // simple comment
-                fn test_simple_comment(self) {
-                    self.inner.test_simple_comment()
-                }
-
-                #[inline]
-                // multiple
-                // simple
-                // comments
-                fn test_multiple_simple_comments(self) {
-                    self.inner.test_multiple_simple_comments()
-                }
-
-                #[inline]
-                /// doc comment
-                fn test_doc_comment(self) {
-                    self.inner.test_doc_comment()
-                }
-
-                #[inline]
-                /// multiple
-                /// doc
-                /// comments
-                fn test_multiple_doc_comments(self) {
-                    self.inner.test_multiple_doc_comments()
-                }
-
-                #[inline]
-                //! bang comment
-                fn test_bang_comment(self) {
-                    self.inner.test_bang_comment()
-                }
+#[proc_macro]
+pub fn delegate(tokens: TokenStream) -> TokenStream {
+    let delegator: DelegatedSegment = syn::parse_macro_input!(tokens);
+
+    let delegator_attribute = delegator.delegator;
+    let functions = delegator.methods.iter().map(|fun| {
+        let input = &fun.method;
+        let signature = &input.sig;
+        let inputs = &input.sig.decl.inputs;
+        let attrs = &input.attrs;
```

Comment:
Maybe add `#[inline(always)]` to `attrs` (fancy customization: unless it already contains `#[inline(never)]`.